### PR TITLE
Several fixes for perfect filters handling

### PR DESCRIPTION
**Pull request by firscity**
This series contains patches related to some problems with perfect filters handling:
1. Correct value for max amount of mapped PF per cascade filter;
2. Fix for PF leak during cascade filter freeing;
3. Fix for PF leak in case of cascade filter setup failure.

### CODE REVIEW

**Review by LKomaryanskiy — APPROVED** [review]
Good job. Approved.

File: drivers/net/ethernet/renesas/rswitch.c [hunk 1]
```diff
@@ -2220,40 +2220,48 @@ static enum pf_type rswitch_get_pf_type_by_num(int num)
...
+		pf_used++;
+	}
 
-		type = rswitch_get_pf_type_by_num(pf_num);
+	/* Disable and free cascade filter */
+	rs_write32(RSWITCH_PF_DISABLE_FILTER, param->priv->addr + FWCFCi(param->pf_cascade_index));
+	clear_bit(param->pf_cascade_index, param->priv->filters.cascade);
+
+	/* Free all used perfect filters */
+	for (i = 0; i < pf_used; i++) {
```
Comment (Deedone) [inline drivers/net/ethernet/renesas/rswitch.c:2242]: Could we have pf_nums filled unevenly (with holes)? judging by the code above it looks like at can happen and we will skip some filters then.

Comment (firscity) [reply drivers/net/ethernet/renesas/rswitch.c:2242]: rswitch_setup_pf() initializes it evenly, so such case is not possible. But just to be sure I will change remembering loop

**Comment by Deedone** [comment]
LGTM +1

**Review by Deedone — APPROVED** [review]

**Review by LKomaryanskiy — APPROVED** [review]

**Pull request merged**